### PR TITLE
Add ability to use a custom service as factory

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/test/app/config/routing.yml [hunk 1]
```diff
@@ -22,3 +22,12 @@ app_book_filterable_index:
...
     prefix: /v{version}
+
+app_create_book_with_custom_factory:
+    path: /books/create-custom
+    method: [POST]
+    defaults:
+        _controller: app.controller.book:createAction
+        _sylius:
+            factory:
+                method: ["@test.custom_book_factory", "createNew"]
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/test/app/config/routing.yml]: In order to make this test more reliable, could we test a method which doesn't exist in the default factory?

Comment (pjedrzejewski) [reply src/Sylius/Bundle/ResourceBundle/test/app/config/routing.yml]: 👍

**Comment by pjedrzejewski** [comment]
Will do the same for repositories. :) Good idea!

File: src/Sylius/Bundle/ResourceBundle/test/app/config/routing.yml [hunk 2]
```diff
@@ -25,9 +25,9 @@ app_versioned_book:
...
 app_create_book_with_custom_factory:
     path: /books/create-custom
-    method: [POST]
+    methods: [POST]
     defaults:
         _controller: app.controller.book:createAction
         _sylius:
             factory:
-                method: ["@test.custom_book_factory", "createNew"]
+                method: ["expr:service('test.custom_book_factory')", "createNew"]
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/test/app/config/routing.yml]: `createNew` -> `createCustom`

**Comment by pamil** [comment]
Tests are failing only because a minor issue in docs, will fix that directly on `1.2` branch, thank you, Paweł! 🎉

**Pull request merged**